akka-http/src/main/scala/akka/http/javadsl/server/directives/PathDirectives.scala

```diff
@@ -1,112 +1,42 @@
-   * The matcher has to match a prefix of the remaining path.
-   * If matched the value extracted by the PathMatcher is extracted on the directive level.
-   *
-   * Each of the arguments  must either be an instance of [[PathMatcher]] or a constant String
-   * that will be automatically converted using `PathMatcher.segment`.
-   */
-  @varargs
-  def rawPathPrefix(matchers: AnyRef*): Directive =
-    RawPathPrefixForMatchers(convertMatchers(matchers))
-
-  /**
-   * Checks whether the unmatchedPath of the [[RequestContext]] has a prefix matched by the
-   * given PathMatcher. However, as opposed to the `pathPrefix` directive the matched path is not
-   * actually "consumed".
-   *
-   * Each of the arguments  must either be an instance of [[PathMatcher]] or a constant String
-   * that will be automatically converted using `PathMatcher.segment`.
-   */
-  @varargs
-  def rawPathPrefixTest(matchers: AnyRef*): Directive =
-    RawPathPrefixTestForMatchers(convertMatchers(matchers))
+import akka.http.scaladsl.model.StatusCodes.Redirection
 
-  /**
-   * Applies the given PathMatchers to a suffix of the remaining unmatchedPath of the [[RequestContext]].
-   * If matched the value extracted by the PathMatchers is extracted and the matched parts of the path are consumed.
-   * Note that, for efficiency reasons, the given PathMatchers must match the desired suffix in reversed-segment
-   * order, i.e. `pathSuffix("baz" / "bar")` would match `/foo/bar/baz`!
-   *
-   * Each of the arguments  must either be an instance of [[PathMatcher]] or a constant String
-   * that will be automatically converted using `PathMatcher.segment`.
-   */
-  @varargs
-  def pathSuffix(matchers: AnyRef*): Directive =
-    Directives.custom(RouteStructure.PathSuffix(convertMatchers(matchers)))
-
-  /**
-   * Checks whether the unmatchedPath of the [[RequestContext]] has a suffix matched by the
-   * given PathMatcher. However, as opposed to the pathSuffix directive the matched path is not
-   * actually "consumed".
-   * Note that, for efficiency reasons, the given PathMatcher must match the desired suffix in reversed-segment
-   * order, i.e. `pathSuffixTest("baz" / "bar")` would match `/foo/bar/baz`!
-   *
-   * Each of the arguments  must either be an instance of [[PathMatcher]] or a constant String
-   * that will be automatically converted using `PathMatcher.segment`.
-   */
-  @varargs
-  def pathSuffixTest(matchers: AnyRef*): Directive =
-    Directives.custom(RouteStructure.PathSuffixTest(convertMatchers(matchers)))
+/**
+ * Only path prefixes are matched by these methods, since any kind of chaining path extractions
+ * in Java would just look cumbersome without operator overloads; hence, no PathMatcher for Java.
+ *
+ * Just nest path() directives with the required types, ending in pathEnd() if you want to fail
+ * further paths.
+ */
+abstract class PathDirectives extends ParameterDirectives {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Number of methods in class exceeds 30](https://www.codacy.com/app/vasseurjb/akka/file/2046270054/issues/source?bid=3132306&fileBranchId=3132379#l31)